src/client/mapblock_mesh.cpp

```diff
@@ -1003,6 +1004,164 @@ static void applyTileColor(PreMeshBuffer &pmb)
 	}
 }
 
+/*
+	MapBlockBspTree
+*/
+
+void MapBlockBspTree::buildTree(const std::vector<MeshTriangle> *triangles)
+{
+	this->triangles = triangles;
+
+	nodes.clear();
+
+	// assert that triangle index can fit into s32
+	assert(triangles->size() <= 0x7FFFFFFFL);
+	std::vector<s32> indexes;
+	indexes.reserve(triangles->size());
+	for (u32 i = 0; i < triangles->size(); i++)
+		indexes.push_back(i);
+
+	root = buildTree(v3f(1, 0, 0), v3f(85, 85, 85), 40, indexes, 0);
+}
+
+static const MeshTriangle *findSplitCandidate(const std::vector<s32> &list, const std::vector<MeshTriangle> &triangles)
```

Comment:
Would you please be so nice to add a description to this function? It would be nice to at least get an idea of how this function helps the depth sorting. More helpful comments = easier to understand for maintenance.

Reply:
Done for this function and added a link to the method to BspTree class.